src/cli_args.rs

```diff
@@ -296,33 +296,38 @@ fn die_error<TRes>(result: Result<TRes, ArgParsingError>) -> bool {
 }
 
 fn color(config: &mut AppConfig, args: &mut Peekable<impl Iterator<Item = String>>) -> bool {
-    let arg = args.next().unwrap();
-    if let Some(spec) = args.next() {
-        die_error(parse_color_arg(&spec, config))
+    if let Some(spec) = args.peek() {
+        let parse_result = parse_color_arg(&spec, config);
+        args.next();
+        die_error(parse_result)
     } else {
-        missing_arg(arg)
+        missing_arg(FLAG_COLOR)
     }
 }
 
 fn line_numbers(config: &mut AppConfig, args: &mut Peekable<impl Iterator<Item = String>>) -> bool {
-    args.next();
-    let spec = if let Some(spec) = args.next() {
-        parse_line_number_style(config, Some(&*spec))
+    let maybe_line_number_style = args.peek();
+    let spec = if let Some(spec) = maybe_line_number_style {
+        if spec.starts_with("-") { // next option
+            parse_line_number_style(config, None)
+        } else {
+            let parse_result = parse_line_number_style(config, maybe_line_number_style);
+            args.next();
+            parse_result
+        }
     } else {
         parse_line_number_style(config, None)
     };
     die_error(spec)
 }
 
-fn html(config: &mut AppConfig, args: &mut Peekable<impl Iterator<Item = String>>) -> bool {
+fn html(config: &mut AppConfig) -> bool {
     config.html = true;
-    args.next();
     true
 }
 
-fn debug(config: &mut AppConfig, args: &mut Peekable<impl Iterator<Item = String>>) -> bool {
```

Comment:
I prefer to keep the signatures of all the arguments handlers the same (debug, html, line_numbers...) even if the functions themselves look dull.

Reply:
Odd, but you are the boss. Do you want me to leave that as is with the warning of `args` being unused, or is there any way to silence it? Like a `pragma` or what analog does Rust have